lib/actions/EventLambdaSchedule.js

```diff
@@ -45,73 +45,95 @@ module.exports = function (S) {
+            Action: populatedEvent.config.enabled ? 'lambda:InvokeFunction' : 'lambda:DisableInvokeFunction',
             Principal: 'events.amazonaws.com',
-            Qualifier: _this.evt.options.stage
+            SourceArn: 'arn:aws:events:' + region + ':' + awsAccountId + ':rule/' + ruleName,
+            Qualifier: stage
           };
-          return _this.aws.request('Lambda', 'addPermission', params, _this.evt.options.stage, _this.evt.options.region)
+          return _this.aws.request('Lambda', 'addPermission', params, stage, region)
+            .then(function() {
+               SUtils.sDebug(`${stage} - ${region} - ${ruleName}: added lambda access policy statement`);
+            });
         })
-        .then(function (data) {
-
-          SUtils.sDebug(`Setting lambda ${lambdaArn}:${stage} as target for rule ${ruleName} for lambda ${functionName}`);
-
-          let params = {
-            Rule: ruleName,
-            Targets: [
-              {
-                Arn: lambdaArn + ":" + stage,
-                Id: functionName
-              }
-            ]
-          };
-
-          if (populatedEvent.config.input) {
-            params.Targets[0].Input = JSON.stringify(populatedEvent.config.input);
+        .then(function () {
+
+          if (populatedEvent.config.enabled) {
+            SUtils.sDebug(`Setting lambda ${lambdaArn}:${stage} as target for rule ${ruleName} for lambda ${functionName}`);
+
+            let params = {
+              Rule: ruleName,
+              Targets: [
+                {
+                  Arn: lambdaArn + ':' + stage,
+                  Id: functionName + '_' + stage,
+                  InputPath: _this.evt.options.inputPath
+                }
+              ]
+            };
+
+            if (populatedEvent.config.input) {
+              params.Targets[0].Input = JSON.stringify(populatedEvent.config.input);
+            }
+
+            return _this.aws.request('CloudWatchEvents', 'putTargets', params, stage, region)
+              .then(function() {
+                SUtils.sDebug(`${stage} - ${region} - ${ruleName}: created target "${functionName}:${stage}"`);
+              });
           }
-          return _this.aws.request('CloudWatchEvents', 'putTargets', params, _this.evt.options.stage, _this.evt.options.region);
+          else {
+            SUtils.sDebug(`Removing lambda ${lambdaArn}:${stage} as target for rule ${ruleName} for lambda ${functionName}`);
+
+            let params = {
+              Rule: ruleName,
+              Ids: [ functionName + '_' + stage ]
+            };
+            return _this.aws.request('CloudWatchEvents', 'removeTargets', params, stage, region)
+              .then(function() {
+                 SUtils.sDebug(`${stage} - ${region} - ${ruleName}: removed existing target "${functionName}:${stage}"`);
+              })
+              .catch(function(error) {})
+          }
+
```

Comment:
Why do we not use an intermediate configuration storage? E.g. for event subscriptions in functions (Kinesis streams, DynamoDB streams, etc.) the function deploy puts the event GUID into the variables file for the deployed region/stage and reads it back if the event has to be recreated or deleted. We just could do the same here and store the GUID (or any other unique identifier) for the rules into the vars. This also would have the advantage that anyone who checks out the plain sources (like a CI system) would have the right information at hand and can start working properly.
Of course manual interaction with the AWS console and deleting stuff will interfere with that, but that's the case with any manual interference in any place of the deployment path.